frontend/src/pages/NewRunV2.tsx

```diff
@@ -167,38 +181,42 @@ function NewRunV2(props: NewRunV2Props) {
   const [isStartingNewRun, setIsStartingNewRun] = useState(false);
   const [errorMessage, setErrorMessage] = useState('');
   const [isParameterValid, setIsParameterValid] = useState(false);
-  const [isClone, setIsClone] = useState(false);
-  const [isRecurringRun, setIsRecurringRun] = useState(apiRecurringRun !== undefined);
-  const initialTrigger =
-    apiRecurringRun && apiRecurringRun.trigger ? apiRecurringRun.trigger : undefined;
+  const [isRecurringRun, setIsRecurringRun] = useState(cloneOrigin.isRecurring);
+  const initialTrigger = cloneOrigin.recurringRun?.trigger
+    ? cloneOrigin.recurringRun.trigger
+    : undefined;
   const [trigger, setTrigger] = useState(initialTrigger);
   const initialMaxConCurrentRuns =
-    apiRecurringRun && apiRecurringRun.max_concurrency !== undefined
-      ? apiRecurringRun.max_concurrency
+    cloneOrigin.recurringRun?.max_concurrency !== undefined
+      ? cloneOrigin.recurringRun.max_concurrency
       : '10';
   const [maxConcurrentRuns, setMaxConcurrentRuns] = useState(initialMaxConCurrentRuns);
   const [isMaxConcurrentRunValid, setIsMaxConcurrentRunValid] = useState(true);
   const initialCatchup =
-    apiRecurringRun && apiRecurringRun.no_catchup !== undefined
-      ? !apiRecurringRun.no_catchup
+    cloneOrigin.recurringRun?.no_catchup !== undefined
+      ? !cloneOrigin.recurringRun.no_catchup
       : true;
   const [needCatchup, setNeedCatchup] = useState(initialCatchup);
-  const [clonedRuntimeConfig, setClonedRuntimeConfig] = useState<PipelineSpecRuntimeConfig>({});
+
+  const initClonedRuntimeConfig = cloneOrigin.isRecurring
+    ? cloneOrigin.recurringRun?.pipeline_spec?.runtime_config
+    : cloneOrigin.run?.run?.pipeline_spec?.runtime_config;
+  const [clonedRuntimeConfig] = useState<PipelineSpecRuntimeConfig>(
```

Comment:
Do we need `useState` here? If not, we can remove it.

Reply:
Yes, we need it. Otherwise, users are not able to change the parameters.